packages/nodes-base/nodes/Github/Github.node.ts

```diff
@@ -2182,6 +2216,9 @@ export class Github implements INodeType {
 					const asBinaryProperty = this.getNodeParameter('asBinaryProperty', i);
 
 					if (asBinaryProperty === true) {
+						if (Array.isArray(responseData)) {
+							throw new NodeOperationError(this.getNode(), 'File Path is a folder, not a file.');
```

Comment:
@janober there's no change in data returned. Just a better error message when there is a `get` request on a folder (not a file) with `asBinaryProperty` enabled. Without this, the code would (does) error out anyway because there's no `responseData.content` on [L2237](https://github.com/n8n-io/n8n/pull/2752/files#diff-2271b836ca282a11927f0f973b16db542596981180e82ba57d057fcbf68d6e71R2237).

I also checked and confirmed that the objects in the array don't have a `content` property... presumably because the `content` of a folder is this array of files/folders.

Reply:
No change in functionality when `asBinaryProperty` is false, but when it's true and `File Path` is a folder.

Error before:
```
ERROR: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined

TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined
    at Function.from (buffer.js:333:9)
    at Object.execute (/usr/local/lib/node_modules/n8n/node_modules/n8n-nodes-base/dist/nodes/Github/Github.node.js:1889:106)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async /usr/local/lib/node_modules/n8n/node_modules/n8n-core/dist/src/WorkflowExecute.js:447:47
```

Error after:
`ERROR: File Path is a folder, not a file.`

Reply:
Ah yes, my mistake. Thanks a lot for explaining.